python/cudf_polars/tests/test_join.py

```diff
@@ -85,16 +86,19 @@ def test_coalesce_join(left, right, how, nulls_equal, join_expr):
     query = left.join(
         right, on=join_expr, how=how, nulls_equal=nulls_equal, coalesce=True
     )
-    assert_gpu_result_equal(query, check_row_order=how == "left")
+    assert_gpu_result_equal(query, check_row_order=False)
 
 
 def test_left_join_with_slice(left, right, nulls_equal, zlice):
     q = left.join(right, on="a", how="left", nulls_equal=nulls_equal, coalesce=True)
 
     if zlice is not None:
+        if DEFAULT_BLOCKSIZE_MODE == "small" and zlice != (0, None):
+            pytest.skip("Cannot match polars' ordering with multiple partitions.")
```

Comment:
For `left.join(right).slice(...)`, the actual output depends on the row ordering, which I'm calling an implementation detail here.

Reply:
Yes, that's reasonable. It also depends on row ordering for a single partition, so how do we match polars for single partitions? Do we sort afterwards?